### PR TITLE
Add a new fiat onboarding option via MoonPay

### PR DESCRIPTION
**Pull request by dan437**
## Explanation
Add a new fiat onboarding option via MoonPay for these networks: Ethereum, BNB Chain, Polygon, Avalanche, Celo

## Testing
- Wait until prod backend is ready, then a prod URL for MoonPay will open in a new tab
- Open the extension
- Click on the `Buy` button
- Click on `Continue to MoonPay` for each of the supported networks

## Screenshots
Fiat onboarding options for Ethereum:
image

MoonPay:
image

### CODE REVIEW

File: app/images/moonpay.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...,0,12.4521,4.63,59.3782,59.3782,0,0,0,14.2879,1.6762,60.0755,60.0755,0,0,0,14.4475-1.6762M732.0427,290.2A107.7909,107.7909,0,0,1,708.336,281.66a87.43,87.43,0,0,1-19.5561-13.33,81.6748,81.6748,0,0,1-14.687-17.64,83.799,83.799,0,0,1-9.339-21.392,92.1875,92.1875,0,0,1,0-48.6906,82.5359,82.5359,0,0,1,9.339-21.3121,84.44,84.44,0,0,1,14.687-17.64,86.02,86.02,0,0,1,19.5561-13.41,107.9957,107.9957,0,0,1,23.7067-8.5408,121.5034,121.5034,0,0,1,53.9588,0,104.73,104.73,0,0,1,23.5471,8.5408,90.9566,90.9566,0,0,1,19.5561,13.41,83.988,83.988,0,0,1,14.9265,17.64,80.9909,80.9909,0,0,1,9.4188,21.3121,92.19,92.19,0,0,1,0,48.6906,82.2161,82.2161,0,0,1-9.4188,21.392,81.2637,81.2637,0,0,1-14.9265,17.64,92.5378,92.5378,0,0,1-19.5561,13.33A104.8355,104.8355,0,0,1,786.0813,290.2a121.8583,121.8583,0,0,1-54.0386,0M372.9434,53.7714h47.8924L502.7318,178.93,584.9472,53.7714H632.84V288.4442H585.4261V127.5257L517.4189,231.2926H488.3641L420.3569,127.5257V288.4442H372.9434Z"/></g><path id="MP_EM_LOGO" class="cls-1" d="M263.5879,95.7848a47.8924,47.8924,0,1,0-47.8924-47.8924,47.8894,47.8894,0,0,0,47.8924,47.8924M116.7378,311.48A116.7378,116.7378,0,1,1,233.4755,194.7426,116.7343,116.7343,0,0,1,116.7378,311.48"/></svg>
```
Comment (georgewrmarshall) [inline app/images/moonpay.svg:1]: Would be good to add this as a logo component like these [#13953](https://github.com/MetaMask/metamask-extension/pull/13953) so it works with darkmode

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. Could add the moonpay svg as a component so it will work with darkmode.

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Looking good! I've made some updates in [this branch on this commit](https://github.com/MetaMask/metamask-extension/commit/9d397e3e158d0b3deb9d8a08db9e23839762a649) as some suggestions. Otherwise LGTM. I haven't tested the functionality as I don't have access to the api key but once those changes are added it's good from a UI perspective! Thanks @dan437

File: ui/components/ui/logo/index.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.logo-moonpay {
```
[the remark this answers is not in the document]
Comment (dan437) [reply ui/components/ui/logo/index.scss:1]: Why not to have default values here and overwrite it in consuming components or via props if needed?

[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/ui/logo/index.scss:1]: We could have it as default props. But that is what `size` which updates width is supposed to take care off.

File: ui/components/ui/logo/logo-moonpay.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const COLOR_MOONPAY_CIRCLES = '#7D00FF';
+
+const LogoMoonPay = ({
+  className,
+  size,
+  color = 'var(--color-text-default)',
+}) => {
+  return (
+    <svg
+      fill={color}
```
Comment (georgewrmarshall) [inline ui/components/ui/logo/logo-moonpay.js:13]: Oh this is my bad sorry! I think `fill={color}` just needs to go in the path that the color concerns

```suggestion
```

**Review by georgewrmarshall — COMMENTED** [review]
Looks great! One tiny change that was my fault to begin with so apologies for that. Once that change is up I'll approve!!

File: ui/components/ui/logo/logo-moonpay.js (hunk 3 above)
Comment (georgewrmarshall) [reply ui/components/ui/logo/logo-moonpay.js:13]: Removed on my local still looks good.

Screen Shot 2022-03-16 at 7 38 58 AM

**Review by georgewrmarshall — APPROVED** [review]
UI LGTM!! Please also get a dev to review the api implementation. Nice work 💯

**Review by darkwing — APPROVED** [review]

**Pull request merged**